docker/dispatcher-image/startup-dispatcher.sh

```diff
@@ -28,15 +28,9 @@ tar -xvzf ${WORK}/src.tar.gz -C ${WORK}/src
 
 # Set up credentials locally as cloud metadata service does not scale.
 credentials_file=${WORK}/creds.json
-iam_account=`gcloud config get-value account`
-gcloud iam service-accounts keys create ${credentials_file} \
-    --iam-account=${iam_account}
+PYTHONPATH=${WORK}/src python3 \
```

Comment:
I don't know if we can set GOOGLE_APPLICATION_CREDENTIALS once we've already imported and started using google cloud libraries. So to be safe, I get this key using a script and then start the dispatcher.

Reply:
i think it works, i remember it checks for auth after the first call. but it is fine if you want to do it this way.

Reply:
First call to an API library? That's problematic because we want to use logging in the key module.
THe only way it would work is if it didn't do any caching of the key at all, so after using the library in the default manner, we could switch to the safe way by setting the env var within the process (this is particularly dangerous if we create any multiprocessing pools on the file level, since those processes wont have any env vars we set after).

Reply:
i mean it happens on google.auth.default() call, not on init.